test/test_lsysparameter.py

```diff
@@ -0,0 +1,143 @@
+from openalea.lpy import *
+from openalea.lpy.lsysparameters import *
+from openalea.plantgl.all import NurbsCurve2D, BezierCurve2D, Polyline2D, NurbsPatch, Material, Texture2D, ImageTexture
+
+def test_param_creation():
+    lp = LsystemParameters()
+    lp.set_color(0,Material((255,20,1)))
+    texture = Texture2D(ImageTexture('image/bjunipc.png'),baseColor=(255,20,1,0))
+    lp.set_color(1,texture)
+    lp.set_option('Module declaration',1)
+    lp.set_option('Warning with sharp module',0)
+    lp.add('test', 1, category='test')
+    lp.add('testFloat', 1.2, category='test')
+    lp.add('testbool',True,category='test2')
+    lp.add('testcurve1',NurbsCurve2D([(0,0,1),(0,1,1),(1,1,1),(1,0,1)]), category='test')
+    lp.add('testcurve2',BezierCurve2D([(0,0,1),(0,1,1),(1,1,1),(1,0,1)]), category='test')
+    lp.add('testcurve3',Polyline2D([(0,0),(0,1),(1,1),(1,0)]), category='test2')
+    lp.add_function('testfunc',NurbsCurve2D([(0,0,1),(0,1,1),(1,1,1),(1,0,1)]), category='test2')
+    lp.add('testpatch',NurbsPatch([[(0,-0.5+j/3.,i/3.,1) for j in range(4)] for i in range(4)]), category='test2')
+    lp.category_info('test')['visible'] = True
+    lp.category_info('test2')['visible'] = True
+    lp.check_validity()
+    return lp
+
+def test_json_validate():
+    lp = test_param_creation()
+    p = lp.generate_json_parameter_dict()
+    assert LsystemParameters.is_valid_schema(p)
+
+def test_json_retrieve():
+    lp = test_param_creation()
+    p = lp.generate_json_parameter_dict()
+    lp2 = LsystemParameters()
+    lp2.retrieve_from_json_dict(p)
+    lp2.check_validity()
+
+    lp.check_similarity(lp2)
+
+def test_json_dump():
+    lp = test_param_creation()
+    p = lp.dumps()
+    lp2 = LsystemParameters()
+    lp2.loads(p)
+    lp2.check_validity()
+
+    lp.check_similarity(lp2)
+
+def test_param_get():
+    lp = test_param_creation()
+    assert 'test' in lp.categories()
+    assert 'test2' in lp.categories()
+    assert len(lp.category_parameters('test')) == 4
```

Comment:
I suggest returning an empty list if a category does not exist:
assert len(lp.category_parameters('does-not-exist')) == 0

Reply:
MAde in aebb2a46757c8a483299abc75050d857fc638146